### ISSUE TITLE
Error: No controllable devices found     at getDeviceList.then.devices (/usr/local/lib/node_modules/homebridge-chamberlain/src/api.js:71:31)

### ISSUE THREAD

**Issue by srector13** [iRayanKhan/homebridge-chamberlain#54]
This is the error I am getting:

[10/18/2019, 4:33:55 PM] [Garage Door] Error: No controllable devices found
    at getDeviceList.then.devices (/usr/local/lib/node_modules/homebridge-chamberlain/src/api.js:71:31)
    at process._tickCallback (internal/process/next_tick.js:68:7)
   hoobs : TTY=pts/0 ; PWD=/var/lib/homebridge ; USER=root ; COMMAND=/bin/journalctl -o cat -n 500 -f -u homebridge -u homebridge-config-ui-x
pam_unix(sudo:session): session opened for user root by (uid=0)

It was working fine earlier today. I have not changed anything

"accessories": [
        {
            "accessory": "Chamberlain",
            "name": "Garage Door",
            "username": "username@me.com",
            "password": "MyPassword"
        }
    ],

**Comment by mhfen** [comment]
I am also experiencing this issue. It was also working fine earlier today. The first time I noticed it not working was about 12 pm MDT. 

Could be an API change on the Chamberlain side of things?

**Comment by ycao56** [comment]
I am running into same issue

**Comment by ironScripter** [comment]
Having the same issue as you guys.

**Comment by brandoniske** [comment]
Related to issue [https://github.com/caseywebdev/homebridge-chamberlain/issues/53](url) when I removed the "deviceId" config parameter. When I specify deviceId, I don't get an error, but no functionality.

**Comment by smallen13** [comment]
I just had this problem today, as well. Chamberlain app still works.

**Comment by mokrol** [comment]
Same here.

**Comment by jocamero** [comment]
Same issue.

**Comment by codea83** [comment]
I'm running into this issue too. Worked fine before today.

**Comment by Barrow-Smith** [comment]
Same issue here. Just started out of nowhere last night.

**Comment by smallen13** [comment]
There are a few comments here [https://github.com/luisiam/homebridge-liftmaster2/issues/63] that  are suggesting a V4 API was disabled and apps need to use V5. I presume that is the same problem here.

**Comment by graanco** [comment]
shame is any one working. on a fix?

**Comment by deanliou** [comment]
> There are a few comments here [https://github.com/[luisiam/homebridge-liftmaster2/issues/63](https://github.com/luisiam/homebridge-liftmaster2/issues/63)] that are suggesting a V4 API was disabled and apps need to use V5. I presume that is the same problem here.

I came here like everyone else. At first I thought I was having wifi or homebridge problems. MyQ app still works, but this homebridge plugin doesn't anymore. I get this error:

`[10/19/2019, 10:50:23 AM] [Garage Door] Error: No controllable devices found at getDeviceList.then.devices (/usr/lib/node_modules/homebridge-chamberlain/src/api.js:71:31) at process._tickCallback (internal/process/next_tick.js:68:7)`

Prior to that, I was just getting nothing in homebridge so I restarted my homebridge VM, which is when these errors stared showing up.

I figured I might as well look around this Saturday morning if someone had posted a fix :) I would like to help, but I don't know how to set a development environment for any of this stuff. Whoever does contribute will be helping out a whole bunch of us folks. Thank you to whoever can help!

**Comment by ThomasTChan** [comment]
Started receiving this error myself.  Found the issue to be the User-Agent field in the header.  If User-Agent is left as previous value, it does not return a SecurityToken.  Instead it returns:

{
    "SecurityToken": "",
    "ReturnCode": "0",
    "ErrorMessage": "please contact customer care, supportID: 16366184268230220087",
    "CorrelationId": ""
}

resulting in no SecurityToken and unauthorized calls to PutDeviceAttribute

Either wait for the PR merge, or edit your src/api.js @ line 19 and replace:

'User-Agent': 'Chamberlain/3.61.1 (iPhone; iOS 11.4.0; Scale/2.00)',
with
'User-Agent': 'N/A'

Fixed issue in PR-55:  https://github.com/caseywebdev/homebridge-chamberlain/pull/55

**Comment by Barrow-Smith** [comment]
@ThomasTChan thanks for the fix. I made the change in my src/api.js and that solved it.  Garage door working via homekit again.

**Comment by ThomasTChan** [comment]
@Barrow-Smith glad it helped.  

Seems Chamberlain is making changes to their API again, which by the way is a bit awkwardly designed and not RESTful.

**Comment by deanliou** [comment]
@ThomasTChan - Thanks! Made the change manually with nano in my local src/api.js and it's working for me again. Makes me wonder if they might shut down "N/A" eventually since they're inspecting user agents like that. In the meantime, I'll take what I can get! Thanks again.

**Comment by ldhall1856** [comment]
Hi,

I'm having the same issue. 

I hate being ignorant about this stuff,  i just don't have time learn as much about programming as i would like.  

Will this be fixed by a update soon or can someone explain in layman terms how to fix it?  

Thanks in advance.

**Comment by deanliou** [comment]
> Hi,
> 
> I'm having the same issue.
> 
> I hate being ignorant about this stuff, i just don't have time learn as much about programming as i would like.
> 
> Will this be fixed by a update soon or can someone explain in layman terms how to fix it?
> 
> Thanks in advance.

Hi @ldhall1856 - I understand. When I first set up my homebridge setup, I worried that I wouldn't be able to "maintain" the system. I still feel that way. Maybe this will help get you back up and running again, thanks to Thomas' hint: Can you get to a command prompt on your homebridge host? If so, then you only need to do the following:

1. at your command prompt, run `nano /usr/lib/node_modules/homebridge-chamberlain/src/api.js` 

2. barring any permissions restrictions, that will launch nano text editor. arrow-down to around line 17, 18, or 19... where it starts with `'User-Agent': 'Chamberlain xxxxxxx'`

3. Change the text of that line to exactly this: `'User-Agent': 'N/A',`   (don't forget the trailing comma)

4. hit Ctrl-X to exit, answer "y" and press enter to save and exit.

Restart your VM/RPi/Whatever, cross your fingers and hope it starts working again.

Hopefully this "fix" will be more permanent than I think it will be temporary. But hey, we'll take what we can get, right? I've really gotten used to having my garage door connected with siri. Would be nice to come home again to a "smart" garage door Monday after work.

**Comment by mokrol** [comment]
If for some reason it is not in that path (like it was for me) try searching for it with `sudo find / -name api.js` find the correct path and use the instructions from above. 

@deanliou - Thank you. That fixed it for me. Appreciate you posting this.

**Comment by graanco** [comment]
Thanks that fixed mine too...

**Comment by ldhall1856** [comment]
I can't find the folder.  I'm running homebridge in docker and can't seem to get to that file

**Comment by ldhall1856** [comment]
I was able to make the change and it still doesn't work for me.

**Comment by ldhall1856** [comment]
Just got it!

Thank you!

**Comment by smallen13** [comment]
The N/A trick worked for me. Moving to the v5 api is clearly the right answer long(er)-term.

I'm guessing it solves issue #53 as well.

**Comment by dswee999** [comment]
@ldhall1856 what did you do to get it to work on Docker. I’m not having any luck. Thanks!!

**Comment by tarbychark** [comment]
Made the change then restarted homebridge and the garage door still shows as not responding. Working fine in the MyQ app.

**Comment by iRayanKhan** [comment]
If anyone hasn't attempted anything, and would like to try my V4 Api fix, try #57 and let me know if it works for y'all. If not please attach logs so I can try to fix it, before upgrading to the V5 api.

**Comment by pb4ugoout** [comment]
> @ldhall1856 what did you do to get it to work on Docker. I’m not having any luck. Thanks!!

The docker container doesn't include nano. You could install it but vi is already installed. Navigate to the aforementioned folder. Then use vi api.src to open the file in vi. Press i to edit. Use the arrow keys to get to the line you want to change and change it. Then press ESC to enter back into command mode. Then press shift zz to save and exit. Then restart the container.

**Comment by pb4ugoout** [comment]
Changing the user-agent to N/A worked for me.

**Comment by JamesCaudill** [comment]
Changing the user-agent to N/A worked for me also.

**Comment by dswee999** [comment]
@pb4ugoout thanks! Got it working!

**Comment by smallen13** [comment]
For what it's worth, after updating to 1.0.5, my device ID changed, so I had to update that. When I ran homebridge without the device ID to get the new one, I mysteriously have two of them now. The first one was the garage door, but the second one seems to be nothing. It added to homebridge, but threw this error when I toggled it:

`[10/21/2019, 10:08:14 PM] [other device id] Error: invalid response, got HTTP 500: {"code":"500.000","message":"An error occurred while processing the request.","description":"Server Error"}
    at res.text.then.body (/usr/lib/node_modules/homebridge-chamberlain/src/api.js:27:15)
    at process._tickCallback (internal/process/next_tick.js:68:7)`

I was hoping it was the light as an independent switch, but it was not. If anyone figures that out, let me know.

**Comment by AdamSnow** [comment]
@smallen13 You aren't alone... I have two garage door openers and am showing four device IDs. A bit of trial and error to find out what ones were the doors, and what ones just threw the error. 

I too, was hoping for the light!

**Comment by schmidt660** [comment]
Thanks for the comments above. My door stopped working a few days ago via HomeKit and it’s been driving me mad! New to coding / how to do I fix or access the file to amend the code via a home bridge docker on a synology nas? Thanks in advance! Alex

**Comment by nicholastripp** [comment]
Many thanks to all for taking the time to solve and post the solution. Saved me many hours of troubleshooting.

**Comment by iRayanKhan** [comment]
@schmidt660 just update the plugin as a PR has been merged implementing a fix.

**Comment by schmidt660** [comment]
> @schmidt660 just update the plugin as a PR has been merged implementing a fix.

Thanks. It says it is up to date (last updated a few days ago). Still not working via homekit (not responding) - but works fine through the Chamberlain app. 

It shows as working on via the homebridge admin page under ‘Accessories’ however, which is odd?

**Comment by iRayanKhan** [comment]
@schmidt660 sounds like a new issue, can you make a new one so I can keep it organized?

**Comment by iRayanKhan** [comment]
If anyone here has any issues, please make a new issue.

**Comment by schmidt660** [comment]
Thanks - have (hopefully) creates a new issue. First time using GitHub so all new to me!

**Comment by derek-miller** [comment]
For what it’s worth, when I hit this issue I went into the chamberlain web UI and reset my password, updated my homebridge config with the new password and all has been fine since. I didn’t have to update or change any code.

**Comment by schmidt660** [comment]
Thanks- just tried that but still not working

**Comment by ssllii** [comment]
I have version 1.0.6:
 sudo npm show homebridge-chamberlain version
1.0.6

But I am still getting this error:
homebridge[13453]: at getDeviceList.then.devices (/usr/lib/node_modules/homebridge-chamberlain/src/api.js:71:31)
Oct 27 16:50:34 sampi homebridge[13453]: at process._tickCallback (internal/process/next_tick.js:109:7)

I haven't manually updated api.js to use "N/A" yet, do I need to with version 1.0.6?  

And something strange, the files in homebridge-chamberlain/src all have a "1985" date!  When I "touch" it, then it switched back to the proper date.
